src/components/learnDetail/ScriptEdit.tsx

```diff
@@ -3,18 +3,47 @@ import { Script } from '@src/services/api/types/learn-detail';
 import HighlightModal from './HighlightModal';
 import styled from 'styled-components';
 import { COLOR } from '@src/styles/color';
-
+import { api } from '@src/services/api';
 interface ScriptEditProps {
+  scriptsId: number;
   scripts: Script[];
   isHighlight: boolean;
   isSpacing: boolean;
 }
 
 function ScriptEdit(props: ScriptEditProps) {
-  const { scripts, isHighlight, isSpacing } = props;
+  const { scriptsId, scripts, isHighlight, isSpacing } = props;
   const [highlightAlert, setHighlightAlert] = useState<boolean>(false);
+  const [firstLineId, setFirstLineId] = useState<number>();
+  const [order, setOrder] = useState<number>();
+  const [resText, setResText] = useState<string>();
   const learnRef = useRef<HTMLDivElement>(null);
 
+  useEffect(() => {
+    (async () => {
+      if (order && resText && scriptsId) {
+        await api.learnDetailService.postSentenceData(
+          {
+            order: order,
+            text: resText,
+          },
```

Comment:
```suggestion
          {
            order,
            text: resText,
          },
```

단축 속성명이라는 개념이 있습니다 😊!
[링크 참고](https://velog.io/@hongcoder/%EB%8B%A8%EC%B6%95-%EC%86%8D%EC%84%B1%EB%AA%85-%EB%8B%A8%EC%B6%95-%EB%A9%94%EC%84%9C%EB%93%9C%EB%AA%85-%EA%B3%84%EC%82%B0%EB%90%9C-%EC%86%8D%EC%84%B1%EB%AA%85)

마찬가지로 resText도 이름을 text로 바꾸면 `text: text`로 쓸 필요 없이 한 번만 적어줘도 됩니다!
만약 resText에서 res가 중요한 의미를 갖는 게 아니라면 간결하게 수정해도 될 것 같아요~

Reply:
오와 신기해요! 항상 새롭고 유용한 정보 감사합니당~

Reply:
우왕 알아갑니다!